src/geometries/primitives/parametrizedcurve.jl

```diff
@@ -0,0 +1,45 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    ParametrizedCurve(func, range = (0.0, 1.0))
+
+A parametrized curve is a curve defined by a function `func` that maps a
+(unitless) parameter `t` in the given `range` to a `Point` in space.
+
+## Examples
+
+```julia
+ParametrizedCurve(t -> Point(cos(t), sin(t)), (0, 2π))
+```
+"""
+struct ParametrizedCurve{M<:Manifold,C<:CRS,F<:Function,R<:Tuple} <: Primitive{M,C}
+  func::F
+  range::R
+
+  function ParametrizedCurve(func, range=(0.0, 1.0))
+    a, b = promote(range...)
+    r = (a, b)
+    p = func(a)
+    M = manifold(p)
+    C = crs(p)
+    new{M,C,typeof(func),typeof(r)}(func, r)
+  end
+end
```

Comment:
For consistency with other constructors:
```suggestion
struct ParametrizedCurve{M<:Manifold,C<:CRS,F<:Function,R<:Tuple} <: Primitive{M,C}
  func::F
  range::R
  ParametrizedCurve{M,C}(fun::F, range::R) where {M<:Manifold,C<:CRS,F<:Function,R<:Tuple} = new{M,C,F,R}(fun, range)
end

function ParametrizedCurve(fun, range=(0.0, 1.0))
  a, b = promote(range...)
  r = (a, b)
  p = fun(a)
  ParametrizedCurve{manifold(p),crs(p)}(fun, r)
end
```